hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java

```diff
@@ -482,40 +478,6 @@ public static RexNode inlineExpression(List<RexNode> inlinedExpressions, RexNode
             public RexNode visitInputRef(RexInputRef inputRef) {
                 return inlinedExpressions.get(inputRef.getIndex());
             }
-
-            @Override
-            public RexNode visitLocalRef(RexLocalRef localRef) {
-                return localRef;
-            }
-
-            @Override
-            public RexNode visitCall(RexCall call) {
```

Comment:
Superclass method is a bit different:

```java
  @Override public RexNode visitCall(final RexCall call) {
    boolean[] update = {false};
    List<RexNode> clonedOperands = visitList(call.operands, update);
    if (update[0]) {
      // REVIEW jvs 8-Mar-2005:  This doesn't take into account
      // the fact that a rewrite may have changed the result type.
      // To do that, we would need to take a RexBuilder and
      // watch out for special operators like CAST and NEW where
      // the type is embedded in the original call.
      return call.clone(call.getType(), clonedOperands);
    } else {
      return call;
    }
  }
```

The original code always create a clone, the code of a superclass does not. Is it ok?

Reply:
I think it is, it's even better.